client/packages/system/src/Immunisation/VaccineCourseView/VaccineCourseView.tsx

```diff
@@ -121,72 +98,35 @@ const Row = ({
   </Box>
 );
 
-const createNewVaccineCourse = (seed?: any | null): any => ({
-  id: FnUtils.generateUUID(),
-  name: '',
-  description: '',
-  ...seed,
-});
-
-interface UseDraftVaccineCourseControl {
-  draft: any;
-  onUpdate: (patch: Partial<any>) => void;
-  onSave: () => Promise<void>;
-  isLoading: boolean;
-}
-
-// dummy vaccine items
-const VaccineOptions = [
-  {
-    label: 'vaccine 1',
-    value: 'vaccine 1',
-  },
-  {
-    label: 'vaccine 2',
-    value: 'vaccine 2',
-  },
-];
-
-const useDraftVaccineCourse = (): UseDraftVaccineCourseControl => {
-  const [vaccine, setVaccineCourse] = useState<any>(() =>
-    createNewVaccineCourse(seed)
-  );
-
-  const onUpdate = (patch: Partial<any>) => {
-    setVaccineCourse({ ...vaccine, ...patch });
-  };
-
-  const onSave = async () => {
-    console.info('TODO update vaccine course mutation');
-  };
+export const VaccineCourseView: FC = () => {
+  const { setSuffix, navigateUpOne } = useBreadcrumbs();
+  const t = useTranslation('coldchain');
+  const { id } = useParams();
+  const {
+    draft,
+    update: { update },
+    updatePatch,
+    query: { data, isLoading },
+    isDirty,
+  } = useVaccineCourse(id);
+  const { data: demographicData } = useDemographicIndicators();
 
-  const isLoading = false;
+  const [buffer, setBuffer] = useState(draft?.doses ?? 1);
+  const [value, setValue] = useState(draft?.doses ?? 1);
 
-  return {
-    draft: vaccine,
-    onUpdate,
-    onSave,
-    isLoading,
+  const defaultRow: VaccineCourseScheduleNode = {
+    doseNumber: 1,
+    id: FnUtils.generateUUID(),
+    label: t('label.new-row'),
+    __typename: 'VaccineCourseScheduleNode',
   };
-};
-
-export const VaccineCourseView: FC = () => {
-  const { setSuffix } = useBreadcrumbs();
-  const t = useTranslation('coldchain');
-  const { draft, onUpdate, isLoading } = useDraftVaccineCourse();
-  const [buffer, setBuffer] = useState(draft?.numberOfDoses ?? 1);
-  const [value, setValue] = useState(draft?.numberOfDoses ?? 1);
 
   const tryUpdateValue = (value: number | undefined) => {
     if (value === undefined) return;
     const isValid = Number.isInteger(value) && value >= 0 && value <= 10;
```

Comment:
You should be able to pass in these constraints as props on the NumericTextInput rather than having to validate here! I think NumericTextInput should also handle the buffer/value situation for you? You just need to handle undefined

Reply:
Nice OK I'll have a play. I was trying to get this working without this, but couldn't get it working as I wanted.

Reply:
Ah ok - we can have a look together on Monday?

Reply:
Yes sounds great!